### DISCUSSION TITLE
PWA not working with dynamic slug routes

### DISCUSSION THREAD

**Discussion by Somesh-Debnath** [vercel/next.js#44852] [Help] [answered]
Hi, I am trying to make an existing nextjs app a PWA using 'yarn build' but during build time it's giving me this error

```
info  - Generating static pages (6/6)

> Build error occurred
Error: Export encountered errors on following paths:
        /MyList
        /Popular
        /[Search]: /any
        /_error: /404
        /_error: /500
        /login

```

I have a page as a slug /pages/[Search.tsx]  which dynamically displays results and url on search
Here's my next.config file

```
const nextConfig = {
  reactStrictMode: true,
  swcMinify: true,
  compiler: {
    removeConsole: process.env.NODE_ENV !== "development",
  },
  images: {
    domains: ['image.tmdb.org','rb.gy'],
  },
   output: 'standalone',
};

const withPWA = require("next-pwa")({
  dest: "public",
  disable: process.env.NODE_ENV === "development",
  register: true,
});

module.exports = withPWA(nextConfig);

```

```
"scripts": {
    "dev": "next dev",
    "build": "next build && next export",
    "start": "next start"
  },
```

Will anyone please help me resolve the error?

**Comment by icyJoseph** [comment]
Alright, so first thing first, it is not necessarily the build that's failing, but the export step. Let me be clear `next build` is working fine, but you are also doing `next export` and calling the combined process "build".

Now why would export fail, but build succeed. One has to think it has to do with unsupported features. Export generated a static HTML application, which means that any features that require a server, are simply not possible. 

Are you using any of these? https://nextjs.org/docs/advanced-features/static-html-export#unsupported-features

- Image Optimization (default loader) -> I think this might be the issue, though it is possible to get it working IIRC
- Internationalized Routing
- API Routes
- Rewrites
- Redirects
- Headers
- Middleware
- Incremental Static Regeneration
- fallback: true
- getServerSideProps

**Comment by leerob** [comment, marked as answer]
Hey everyone. We now have updated documentation on building PWAs with Next.js.

https://nextjs.org/docs/app/building-your-application/configuring/progressive-web-apps
